test/content-legacy/inline.html

```diff
@@ -37,6 +37,10 @@
                 });
 
                 it('should remove stamped nodes after content is cleared', function(){
+                    var parent = myEl.parentNode;
+                    // pre-test conditions
+                    expect(parent.children.length).to.be.equal(2);
```

Comment:
Can this be in `beforeEach`?

Reply:
Do you think it's a good practice? I've never seen such a use - assertion in setup? Aren't we going to write tests for tests?

Reply:
I've never seen testing for pre-test conditions in a test either. The fact that the test has to check preconditions looks like a design flaw in the testing framework. Maybe we can do some resetting of the parent container in `beforeEach` that makes this check redundant?

Reply:
We do have such resetting at https://github.com/Juicy/juicy-html/blob/44523a39bbb96e11cc850c2ea447fae2c94505a8/test/content-legacy/inline.html#L31
@alshakero as this was your idea to test preconditions could you elaborate?

Reply:
I'm not a testing expert. But when I see a test that is prone to fool me into a passing test, I try to test before and after the action in question. 

For example, if we want to test `delete`: 
```js
expect(myObj.myVar).toBe(undefined)
```
... is very likely to pass even if `delete` is broken. That's why I usually do:
```js
expect(myObj.myVar).not.toBe(undefined);
delete myObj.myVar;
expect(myObj.myVar).toBe(undefined)
```

And the tests I thought needed a control check are the ones that test for `null` and `children.length == 1`. Checking if `children.count == 2` then `remove(child)` then `children.count == 1` sounds like a more robust test than only `children.count == 1`.

#### However, I'm not asserting and merely suggesting.

Reply:
Well `beforeEach` makes sure we have elements there https://github.com/Juicy/juicy-html/blob/44523a39bbb96e11cc850c2ea447fae2c94505a8/test/content-legacy/inline.html#L31

Doing `expect` is just double checking. The question is whether it is a confusing paranoid mode, or better safe than sorry approach, as `beforeEach` uses `<test-fixture>` which may also have bugs.

Reply:
WDYT @warpech ?

Reply:
You have convinced me, let's keep it :)

Reply:
I wasn't 100% convinced either. I'm fine to keep it, but I'd like to avoid going to paranoid, to make tests easier to read and write.